frontend/src/app/projects/models/backup-settings/view-logs-dialog/view-logs-dialog.component.ts

```diff
@@ -29,26 +29,31 @@ export class ViewLogsDialogComponent {
     this.refreshEvents();
   }
 
-  logs = '';
-  events = '';
+  logs?: string = undefined;
+  events?: string = undefined;
 
   refreshEvents(): void {
     combineLatest([
-      this.projectService.project$.pipe(filter(Boolean)),
-      this.modelService.model$.pipe(filter(Boolean)),
-      this.pipelineService.pipeline$.pipe(filter(Boolean)),
+      this.projectService.project$.pipe(filter(Boolean), take(1)),
+      this.modelService.model$.pipe(filter(Boolean), take(1)),
+      this.pipelineService.pipeline$.pipe(filter(Boolean), take(1)),
       this.pipelineRunService.pipelineRun$.pipe(filter(Boolean)),
     ])
       .pipe(
         untilDestroyed(this),
-        switchMap(([project, model, pipeline, pipelineRun]) =>
-          this.pipelineRunService.getEvents(
+        filter(
+          ([_project, _model, _pipeline, pipelineRun]) =>
+            this.logs === undefined ||
+            !this.pipelineRunService.pipelineRunIsFinished(pipelineRun.status)
```

Comment:
```suggestion
            !(this.logs && this.pipelineRunService.pipelineRunIsFinished(pipelineRun.status))
```

Reply:
First of all, `this.logs === undefined` is need here. `this.logs` could also be an empty array, which has to be handled differently.

Also, your suggestion is different. There is no `!` before the `this.logs` in the origin implementation.